src/pdn/src/PdnGen.tcl

```diff
@@ -5861,31 +5861,35 @@ proc repair_channel {channel layer_name tag min_size} {
     set other_strap [lindex [odb::getRectangles [odb::andSet [odb::newSetFromRect $xMin $yMin $xMax $yMax] $stripe_locs($layer_name,$other_tag)]] 0]
 
     if {[get_dir $layer_name] == "hor"} {
+      set center [expr ($xMax + $xMin) / 2]
       if {$xMax - $xMin < $min_size} {
-        set center [expr ($xMax + $xMin) / 2]
         set xMin [expr $center - $min_size / 2]
         set xMax [expr $center + $min_size / 2]
       }
-      if {[$other_strap yMin] - $channel_spacing - $width > $yMin} {
+
+      set other_strap_center [expr ([$other_strap yMin] + [$other_strap yMax]) / 2]
+      if {($center <= $other_strap_center) && ([$other_strap yMin] - $channel_spacing - $width > $yMin)} {
         # debug "Stripe below $other_strap"
         set stripe [odb::newSetFromRect $xMin [expr [$other_strap yMin] - $channel_spacing - $width] $xMax [expr [$other_strap yMin] - $channel_spacing]]
-      } elseif {[$other_strap yMax] + $channel_spacing + $width < $yMax} {
+      } elseif {($center > $other_strap_center) && ([$other_strap yMax] + $channel_spacing + $width < $yMax)} {
```

Comment:
@Colin-Holehouse Why the comparison is between x coordinate  (center) and y coordinate (other_strap_center)?

Reply:
Perhaps this is the cause of why I see the straps are still improperly spaced.

Reply:
Good spot - seems like this was "correct by accident" in the original testcase.